### ISSUE TITLE
ark and jupyter kernels staying running after quitting zed

### ISSUE THREAD

**Issue by alvinahayward** [zed-industries/zed#21911]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

running R and Python files in cell mode. Quit zed. R language server ark and jupyter kernel still running in process list

### Environment

Zed: v0.165.4 (Zed Preview)
OS: macOS 14.4.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

**Comment by notpeter** [comment]
I can reproduce. Steps:

1. Setup an R environ:
```
brew install --cask r
Rscript --version
Rscript -e 'options(repos = "https://cran.rstudio.com/"); install.packages("languageserver")'
Rscript -e 'options(repos = "https://cran.rstudio.com/"); install.packages("lintr")'
Rscript -e 'packageVersion("languageserver")'
Rscript -e 'packageVersion("lintr")'

cd /tmp
curl -L -o ark-latest-darwin.zip \
    $(curl -s "https://api.github.com/repos/posit-dev/ark/releases/latest" | \
    jq -r '.assets[] | select(.name | contains("darwin-universal") and (contains("debug") | not)) | .browser_download_url')
unzip ark-latest-darwin.zip ark
sudo mv /tmp/ark /usr/local/bin/
```

2. Actual repro:
```
echo "cat ('Hello world!')" > a.R
zed a.R:1
```
Select "Ark" from the kernel dropdown.
Click the ♻️ (or whatever it's called) and "Start REPL for Ark"
```sh
ps aux |grep '[/]usr/local/bin/ark'
```
Quit Zed and it's still running:
```sh
ps aux |grep '[/]usr/local/bin/ark'
```

Rinse and repeat
<img width="1284" alt="Screenshot 2025-01-06 at 16 27 03" src="https://github.com/user-attachments/assets/1aa1df74-54c4-48a8-900d-c4bcea5bfb17" />

I assume this applies similarly to Jupyter kernels.

**Comment by rawhuul** [comment]
Hi there, is there any workaround for this issue? It also occurs when using `IRKernel`, and the processes continue to consume memory in the background.

**Comment by JonGretar** [comment]
> Hi there, is there any workaround for this issue? It also occurs when using `IRKernel`, and the processes continue to consume memory in the background.

No.. They are not focusing on jupyter now until after release 1.0.
Personally I just have a script that prints out a message warning of many jupyter processes when I start a new shell. So at least I get to know about it.
